### ISSUE TITLE
Custom AutoId implementation

### ISSUE THREAD

**Issue by fubar-coder** [litedb-org/LiteDB#1239]
What can I do, if I'd like to provide a custom AutoId implementation, that - for example - uses one of the following:

- Sequential GUID
- A unique string

Do I really **have to** set  the ID property myself? It's sad, that `RegisterAutoId` was removed with no viable replacement.

**Comment by lobster2012-user** [comment]
Hi.
What about ?
```_id = ...```

**Comment by fubar-coder** [comment]
Can you please explain in detail what you mean? `_id` is the field in the database, but I don't know how it would help with a custom AutoId implementation?

**Comment by lobster2012-user** [comment]
You can add this field to your class and fill it.

**Comment by fubar-coder** [comment]
Ugh, that is **exactly** what I don't want to do. The original `RegisterAutoId` was exactly what I'd need.

**Comment by lobster2012-user** [comment]
Yes, I do not see such a possibility not in v4, not in v5 versions. (Registerautoid)
It is difficult to say why the author refused this opportunity.

This can be solved on your own without changes in LiteDB.
With a great desire, add a little sugar.

**Comment by JKamsker** [comment]
Thanks for the suggestion! It's now part of #2837, which collects the related feature requests in one place so they can be discussed and prioritised together. Closing this as a duplicate — please follow #2837 and add your use case there.
